src/hub/ShareClassManager.sol

```diff
@@ -554,30 +563,33 @@ contract ShareClassManager is Auth, IShareClassManager {
         return lastEpoch - userOrder.lastUpdate + 1;
     }
 
+    //----------------------------------------------------------------------------------------------
+    // Internal methods
+    //----------------------------------------------------------------------------------------------
+
     function _updateMetadata(ShareClassId scId_, string calldata name, string calldata symbol, bytes32 salt) private {
         uint256 nameLen = bytes(name).length;
         require(nameLen > 0 && nameLen <= 128, InvalidMetadataName());
 
         uint256 symbolLen = bytes(symbol).length;
         require(symbolLen > 0 && symbolLen <= 32, InvalidMetadataSymbol());
 
-        require(salt != bytes32(0), InvalidSalt());
-
         ShareClassMetadata storage meta = metadata[scId_];
 
-        // Ensure that the salt remains unchanged if it's already set,
-        // or that it is being set for the first time.
-        require(salt == meta.salt || meta.salt == bytes32(0), InvalidSalt());
+        // Ensure that the salt is not being updated or is being set for the first time
+        require(
+            (salt == bytes32(0) && meta.salt != bytes32(0)) || (salt != bytes32(0) && meta.salt == bytes32(0)),
```

Comment:
So one could update with `salt != meta.salt` but `meta.salt != bytes(0)` and `salt != bytes(0)` and it will succeed? Isn't that a bit odd?

Reply:
`updateMetadata` doesn't pass a salt anymore, only on `addShareClass`, so it anyways never can be updated

Reply:
I agree we should maybe change the logic but didn't really see a better way, without updating salt directly in `addShareClass` which would be more expensive